arch/arm64/src/imx9/imx9_lpuart.c

```diff
@@ -1485,96 +1483,82 @@ static int imx9_interrupt(int irq, void *context, void *arg)
   pm_activity(PM_IDLE_DOMAIN, CONFIG_IMX9_PM_SERIAL_ACTIVITY);
 #endif
 
-  /* Loop until there are no characters to be transferred or,
-   * until we have been looping for a long time.
+  /* Get the current UART status and check for loop
+   * termination conditions
    */
 
-  handled = true;
-  for (passes = 0; passes < 256 && handled; passes++)
```

Comment:
I think the purpose of this was to handle more characters during 1 interrupt. The path to get here is quite processing heavy. Although I agree the loop most likely never gets executed on this architecture, the CPU is much faster than what it takes to receive 1 character from the UART.

Reply:
This looping is already done in uart_xmitchars / uart_recvchars in serial_io.c . So it is IMHO unnecessary here.

Reply:
I agree